### ISSUE TITLE
ESP32-S3 hangs after a short period when using the internal serial bridge over native USB (using Arduino framework)

### ISSUE THREAD

**Issue by codewitch-honey-crisis** [espressif/arduino-esp32#8399]
### Board

ESP32-S3 various - M5 Stack Atom for example

### Device Description

This happens on the Lilygo T-Display S3, the M5 Stack Atom, the Lilygo T-QT Pro, and various Makerfabs S3 boards when using native USB serial bridge.

### Hardware Configuration

Yes. An LCD panel - some SPI, some parallel 8 and 16, some RGB interface. There may be other peripherals embedded depending on the kit, but I'm not using them. Just serial and LCD.

### Version

v2.0.6 (other versions as well including 2.0.3

### IDE Name

PlatformIO

### Operating System

Windows 11

### Flash frequency

40Mhz

### PSRAM enabled

yes, on boards that have it. RGB panel boards require it.

### Upload speed

921600-2000000 (depending on CP2102 or native USB serial)

### Description

NOTE: I need to do further research to see if this fails under the ESP-IDF

When running the repo project here, it works flawlessly on devices over standard CP2102 serial UART bridge (like the original ESP32s, or the Makerfabs boards using the TTL UART input) but hangs almost immediately when using native USB serial.

Here's the reproduction project. Requires platform IO, visual studio, and the ability to run a windows .NET app with elevated privileges.

https://www.codeproject.com/Articles/5362592/EspMon-Reboot

Sorry it's not simpler, but I needed serious serial coms, and I didn't want to spend all day trying to reproduce this in simpler code.

The main deal here is the app completely hangs up. I don't know exactly where because obv I am not getting a stack dump, and I don't have the equipment for JTAG on these boards.

### Sketch

```cpp
https://github.com/codewitch-honey-crisis/esp_mon2
```

### Debug Message

```plain
No debug. Total hang.
```

### Other Steps to Reproduce

If you want to reproduce on the makerfabs boards you'll need to search for Serial. and replace with USBSerial. or change the platformIO.ini such that the arduino layer maps it for you.

Run the app on the PC, choose the appropriate platformio cfg entry and upload the firmware to any of the suspect devices. 

### I have checked existing issues, online documentation and the Troubleshooting Guide

- [X] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

**Comment by codewitch-honey-crisis** [comment]
Update: I've confirmed this reproduces under the ESP-IDF as well (see code at github link)

**Comment by codewitch-honey-crisis** [comment]
Right, this is an issue with the Hardware Serial over USB mode. I'll certainly try to convince the Arduino IDE to run C++14 or better which my code presently requires, but I have to do some legwork. PIO doesn't give me 2.0.9

**Comment by codewitch-honey-crisis** [comment]
It's still running. It should have frozen by now if there was a problem. I'm closing this issue as fixed (at least in 2.0.9)

Sorry for the churn. :)
